onvm/onvm_nflib/onvm_common.h

```diff
@@ -325,8 +336,6 @@ struct onvm_nf_info {
         uint16_t time_to_live;
         /* If set NF will stop after pkts TX reach pkt_limit */
         uint16_t pkt_limit;
-        /* Pointer to NF defined state data */
-        void *data;
```

Comment:
~Does this might still need data?~ Probably not, read explanation

For advanced rings one would spawn a pthread, give in the `onvm_nf_init_data` as an argument, and that child will initialize itself but it would lack state data.  Although it should also be on the advanced rings developers themselves to build a tuple struct that has` onvm_nf_init_data` + a `void *data` pointer of some sorts.

Reply:
Although might be a bit cleaner this way, doesn't hurt our API either just coping a pointer inside of onvm_mgr. @nks5295 thoughts?

Reply:
Struct names changed but I'm going with not keeping the data as part of the nf_init_cfg. Reply if you don't agree